### ISSUE TITLE
"•" is pronounced by NVDA/JAWS as "bullet" even when it's invisible

### ISSUE THREAD

**Issue by limonte** [vaadin/vaadin-text-field#221]
The line which causes the issue: https://github.com/vaadin/vaadin-text-field/blob/master/theme/lumo/vaadin-text-field.html#L64

Live demo: https://cdn.vaadin.com/vaadin-date-picker/3.0.0/demo/#date-picker-basic-demos

When focusing the second example 

### Expected 

The label is pronounced as "Pick a date"

### Actual

The label is pronounced as "Pick a date bullet"

**Comment by jouni** [comment]
We probably should have an explicit required-indicator element that we can mark as hidden from screen readers.

That would work a bit towards what I proposed in https://github.com/vaadin/vaadin-text-field/issues/212#issuecomment-380463406

**Comment by jouni** [comment]
Ideally, the required indicator element should have `aria-label="required"` or something else so that the label would be pronounced “Pick a date, required”. Though, that means there needs to be a way to translate that.

**Comment by anezthes** [comment]
There's probably a better way of coding this but this works in Chrome/Safari with VoiceOver:

```
<label for="input">
  ${this.label}
  ${this.required ? html`<span>*</span>` : html``}
</label>
<div id="input-wrapper">
  <slot name="prefix"></slot>
  <input id="input" type="text" value="${this.value}">
...
```

**Comment by jouni** [comment]
Yeah, an explicit element would be better. Keep in mind, though, that the actual indicator should be defined from the style sheet, so that it is controllable with a theme. That also gives to option to choose whether to indicate required or optional fields. 

In the case of optional fields, I think users would want to hear the screen reader announce that as well, for example: “Bio, optional”. Would you agree, or do you think it is enough that the user does not hear “required” announced to imply that the field is optional?

If I remember correctly, required fields get announced already, because of the `required` attribute in the input element.

**Comment by jouni** [comment]
Adding this here as a possible workaround, at least for some users: https://www.stefanjudis.com/today-i-learned/css-content-accepts-alternative-text/

**Comment by web-padawan** [comment]
Closing in favor of https://github.com/vaadin/vaadin-web-components/issues/88 which describes the same problem.
